### PR TITLE
PHPStan (Level 4) on classes/

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | PHPStan (Level 4) on classes/
| Type?             | improvement
| Category?         | CO
| BC breaks?        | yes / no
| Deprecations?     | yes / no
| Fixed ticket?     | Relative to #16471
| How to test?      | Github Actions are :green_circle: & @boubkerbribri tests are :green_circle: 

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26727)

### CODE REVIEW

**Review by matthieu-rolland — DISMISSED** [review]

File: classes/cache/CacheMemcached.php [hunk 1]
```diff
@@ -79,10 +79,7 @@ public function connect()
             $this->memcached->addServer($server['ip'], $server['port'], (int) $server['weight']);
         }
 
-        $versions = $this->memcached->getVersion();
-        $this->is_connected = is_array($versions)
```
Comment (matks) [inline classes/cache/CacheMemcached.php:83]: Are you sure that `$versions` is always an array? Sure sure sure 😄  ?

**Review by matks — DISMISSED** [review]
LGTM, I have just one thing I'm worried about, see above question

File: classes/cache/CacheMemcached.php (hunk 1 above)
Comment (Progi1984) [reply classes/cache/CacheMemcached.php:83]: :tada: https://www.php.net/manual/fr/memcached.getversion.php :tada:

**Comment by Progi1984** [comment]
@boubkerbribri Could you launch your tests on this PR ?

**Comment by Progi1984** [comment]
@matthieu-rolland & @matks I just rebased the PR. Could you reapprove ?

**Review by matthieu-rolland — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]
Tested with UI tests. All good!

**Comment by Progi1984** [comment]
Thanks @boubkerbribri

**Pull request merged**